src/Microsoft.ApplicationInsights/TelemetryClient.cs

```diff
@@ -60,10 +59,12 @@ public TelemetryClient(TelemetryConfiguration configuration)
         /// </summary>
         public TelemetryContext Context
         {
-            get { return LazyInitializer.EnsureInitialized(ref this.context, () => new TelemetryContext()); }
-            internal set { this.context = value; }
+            get;
+            internal set;
         }
 
+        = new TelemetryContext();
```

Comment:
I think you've traded one perf problem for another. From profiling, did you see lock contention or high CPU usage in the old Context getter?
For sure, you're now allocating a new TelemetryContext on each new TelemetryClient. There are some (anti-)patterns where new TelemetryClient instances are created frequently and I think this is what the LazyInitializer was trying to improve.

Reply:
The TelemetryContext is needed in most of the paths for TelemetryClient, thus it is always initialized. We are saving on additional checks and Volatile reads here: https://referencesource.microsoft.com/#mscorlib/system/threading/LazyInitializer.cs

Reply:
Under what conditions TelemetryClients are created and what do they do that they don't need the TelemetryContext. Is the perf critical for such scenarios?

Reply:
@Mikhail-Pranovich we treat `TelemetryClient` as lightweight object that can be used like `(new TelemetryClient()).TrackEvent("Configuration was updated");`. We have quite a few customers who uses it this way.

It is not a use we have in default out of the box case.

So if this thread contention problem is critical for out of the box experience, it may be an OK tradeoff.

Reply:
Those calls end up in Track, which access Context in case if enabled. If not-enabled high perf scenario is critical for those guys, we can advise on re-using telemetry client.